opensearch-operator/pkg/builders/cluster.go

```diff
@@ -162,9 +165,37 @@ func NewSTSForNodePool(
 		//vendor ="elasticsearch"
 	}
 
+	isDataNode := helpers.CheckDataRoles(selectedRoles)
+
 	var jvm string
 	if node.Jvm == "" {
-		jvm = "-Xmx512M -Xms512M"
+		// Get the node memory limit
+		memoryLimit := node.Resources.Limits.Memory()
+		memString := memoryLimit.String()
+		re := regexp.MustCompile("[0-9]")
+		numberString := re.FindString(memString)
+		memInt, _ := strconv.Atoi(numberString)
+
+		// Define regular expressions to check resource unit
+		reG := regexp.MustCompile(`G`)
+		// Convert Gb to Mb
+		if reG.MatchString(memString) {
+			memInt = memInt * 1024
+		}
+
+		// Calculate jvm memory limit by cluster role
+		if isDataNode {
+			newMemory := int(math.Round(float64(memInt) * 0.5))
+			newMemoryStr := strconv.Itoa(newMemory) + "M"
```

Comment:
If the user is configure JVM with Gi or Mi is it also work ?

Reply:
if user set pod resorces with Gi it convert it to mb in any way the jvm gonna be in Mb when the jvm not specified